PlayFabClientSDK/source/PlayFabHttp/PlayFabHttp.cs

```diff
@@ -41,28 +41,28 @@ public class PlayFabJsonSuccess<TResult> where TResult : PlayFabResultCommon
 
     public static class PlayFabHttp
     {
-        private static readonly IPlayFabHttp _http;
-
-        static PlayFabHttp()
-        {
-            var httpInterfaceType = typeof(IPlayFabHttp);
-            var types = typeof(PlayFabHttp).GetAssembly().GetTypes();
-            foreach (var eachType in types)
-            {
-                if (httpInterfaceType.IsAssignableFrom(eachType) && !eachType.IsAbstract)
-                {
-                    _http = (IPlayFabHttp)Activator.CreateInstance(eachType.AsType());
-                    return;
-                }
-            }
-            throw new Exception("Cannot find a valid IPlayFabHttp type");
-        }
+        private static ITransportPlugin _transport;
 
         public static async Task<object> DoPost(string urlPath, PlayFabRequestCommon request, string authType, string authKey, Dictionary<string, string> extraHeaders)
         {
             if (PlayFabSettings.TitleId == null)
                 throw new Exception("You must set your titleId before making an api call");
-            return await _http.DoPost(urlPath, request, authType, authKey, extraHeaders);
+            if (_transport == null)
+                _transport = GetTransport();
+
+            return await _transport.DoPost(urlPath, request, authType, authKey, extraHeaders);
+        }
+
+        private static ITransportPlugin GetTransport()
+        {
+            var transport = (ITransportPlugin)PluginManager.Instance.GetPlugin(PluginContract.Transport);
```

Comment:
I don't think that we should cache a plugin.
If the plugin is replaced, but we've already fetched it once, we won't get the new one.
Let's just assume that GetPlugin is fast (enough).